File: src/main/kotlin/rainbow/inner/function/pointfunction/Hypercube.java
package rainbow.inner.function.pointfunction;

import rainbow.coordinate.point.PointForAxes;
import rainbow.inner.function.MyFunction;
import rainbow.coordinate.graphics.CoordinateGraphics;

/**
 * 任意维度的超方体
 *
 * @author Rainbow Yang
 */
public class Hypercube extends MyFunction {
    private double length;
    private int size;

    public Hypercube(double length, int size) {
        this.length = length;
        this.size = size;
    }

    @Override
    public void paintMain(CoordinateGraphics mg) {
        mg.paintLocation(new PointForAxes(length, size));
    }
}

File: src/main/kotlin/rainbow/coordinate/function/pointfunction/Hypercube.kt
package rainbow.coordinate.function.pointfunction;

import rainbow.coordinate.function.CoordinateFunction
import rainbow.coordinate.function.CoordinateFunctionPainter
import rainbow.coordinate.graphics.CoordinateGraphics
import rainbow.coordinate.point.PointForAxes
import rainbow.coordinate.system.CoordinateSystem

/**
 * 任意维度的超方体
 * 基于paintLocation实现
 * 所以目前只能从原点出发
 * @author Rainbow Yang
 */
class Hypercube(var size: Int, var length: Double = 5.0) : CoordinateFunction() {
    override var coordinateSystem: CoordinateSystem
        get() = super.coordinateSystem
        set(value) {
            painter = object : CoordinateFunctionPainter<Hypercube>(this, value) {
                override fun paintMain(cg: CoordinateGraphics) {
                    cg.paintLocation(PointForAxes(length, size))
                }
            }
        }
}
